### PR TITLE
Added hashing params dict

### PR DESCRIPTION
**Pull request by andrii-ivaniuk**

### CODE REVIEW

File: src/acachecontrol/cache.py [hunk 1]
```diff
@@ -25,23 +28,25 @@ def add(self, key, value):
...
+    def delete(self, key: Tuple[str, str, Dict]) -> None:
+        self.cache.pop(self._make_key_hashable(key), None)
 
-    def clear_cache(self):
+    def clear_cache(self) -> None:
         self.cache = {}
 
-    async def register_new_key(self, key, wait_timeout=10):
+    async def register_new_key(
+        self, key: Tuple[str, str, Dict], wait_timeout: float = 10.0
```
Comment (MasterSergius) [inline src/acachecontrol/cache.py:41]: We'll need to make wait_timeout configurable, but it is ok for now

File: src/acachecontrol/cache.py [hunk 2]
```diff
@@ -51,24 +56,26 @@ async def register_new_key(self, key, wait_timeout=10):
...
-        return (method, url, self._dict_to_tuple(params))
+        return method, url, self._dict_hash(params)
 
-    def _dict_to_tuple(self, input_dict):
-        """Dict is not hashable, thus we need something else."""
-        return tuple(sorted(input_dict.items()))
+    @staticmethod
+    def _dict_hash(input_dict: Dict) -> str:
+        """Get dict hash."""
+        return hashlib.sha256(json.dumps(input_dict).encode()).hexdigest()
```
Comment (MasterSergius) [inline src/acachecontrol/cache.py]: This is not correct:
```py
>>> input_dict = {'a': 1, 'b': 2}
>>> hashlib.sha256(json.dumps(input_dict).encode()).hexdigest()
'd8497d9d82770a70729261095aa98f7ef5154d7af499f8037b6ca250296785a6'
>>> input_dict = {'b': 2, 'a': 1}
>>> hashlib.sha256(json.dumps(input_dict).encode()).hexdigest()
'd5b8aed265256c893850d31609a884cd910a452df090e34ce4d318c9ef5bd771'
```

This should be working:
```py
>>> input_dict = {'a': 1, 'b': 2}
>>> hashlib.sha256(json.dumps(sorted(input_dict.items())).encode()).hexdigest()
'91cea03dbc87dd3dbe06a0399ab693756cb08e6fd217b42780296a954c4cd308'
>>> input_dict = {'b': 2, 'a': 1}
>>> hashlib.sha256(json.dumps(sorted(input_dict.items())).encode()).hexdigest()
'91cea03dbc87dd3dbe06a0399ab693756cb08e6fd217b42780296a954c4cd308'
>>> 

```

**Review by MasterSergius — CHANGES_REQUESTED** [review]
Fix dict hash

File: src/acachecontrol/cache.py (hunk 1 above)
Comment (andrii-ivaniuk) [reply src/acachecontrol/cache.py:41]: what do you mean?

Comment (MasterSergius) [reply src/acachecontrol/cache.py:41]: something like AsyncCacheControl(wait_timeout=120)

**Review by MasterSergius — APPROVED** [review]

**Pull request merged**